tests/Controller/CRUDControllerTest.php

```diff
@@ -3644,6 +3644,114 @@ public function testBatchActionWithConfirmation(): void
         $this->assertSame('@SonataAdmin/CRUD/batch_confirmation.html.twig', $this->template);
     }
 
+    /**
+     * NEXT_MAJOR: Remove this legacy group.
+     *
+     * @group legacy
+     */
+    public function testBatchActionWithConfirmationWithoutAllElements(): void
+    {
+        $batchActions = ['delete' => ['label' => 'Foo Bar', 'translation_domain' => 'FooBarBaz', 'ask_confirmation' => true]];
+
+        $this->admin->expects($this->once())
+            ->method('getBatchActions')
+            ->willReturn($batchActions);
+
+        $data = ['action' => 'delete', 'idx' => ['123', '456']/*, 'all_elements' => false*/];
```

Comment:
Instead of the commented values, I think an explicit comment about what they are relevant will be better.